_posts/2020-09-07-introduction-to-system-tests.md

```diff
@@ -0,0 +1,275 @@
+        assertThat(srv.getSpec().getPorts(), is(Collections.singletonList(kc.createServicePort(KafkaCluster.EXTERNAL_PORT_NAME, KafkaCluster.EXTERNAL_PORT, KafkaCluster.EXTERNAL_PORT, "TCP"))));
+        checkOwnerReference(kc.createOwnerReference(), srv);
+    }
+}
+```
+
+### Integration tests
+
+Next up are the integration tests, which move one level higher.
+You can find all these tests inside our `api` module.
+Before we get into the details again, we should understand the informal definition of integration tests.
+
+> Integration tests determine if independently developed units of software work correctly when they are connected to each other. (Martin Fowler)
+
+In our case, the purpose of integration tests is to ensure that we can create a resource from the POJOs, serialize it and
+create the resource in Kubernetes.
+You should notice that in this level of testing we are using the Kubernetes cluster for validation of custom resources.
+To make this all easy to understand let me clarify it with example:
+
+```java
+@Test
+public void testKafkaWithEntityOperator() {
+    createDelete(Kafka.class, "Kafka-with-entity-operator.yaml");
+}
+
+protected <T extends CustomResource> void createDelete(Class<T> resourceClass, String resource) {
+    T model = loadResource(resourceClass, resource);  // load model in that case Kafka model representation
+    String modelStr = TestUtils.toYamlString(model);  
+    assertDoesNotThrow(() -> createDelete(modelStr), 
+    "Create delete failed after first round-trip -- maybe a problem with a defaulted value?\nApplied string: " + modelStr);
+}
+```
+
+You can see that the test case basically applies the YAML representation of Kafka and expects the creation of a `Kafka` customer 
+resource object in Kubernetes cluster is correct.
+This is applied for every custom resource supported by `Strimzi`, such as `KafkaConnect`, `KafkaMirrorMaker`, `KafkaBridge`,
+`KafkaTopic`, `KafkaUser` and so on.
+
+### System tests
+
+After this level is successfully completed we can talk about our system tests.
+Take a look at the informal definition again:
+
+> System testing is a level of testing that validates the complete and fully integrated software product.
+> The purpose of a system test is to evaluate the end-to-end system specifications.
+> Usually, the software is only one element of a larger computer-based system.
+> Ultimately, the software is interfaced with other software/hardware systems.
+> System Testing is actually a series of different tests whose sole purpose is to exercise the full computer-based system. (Guru)
+
+In our case the system tests validates all components and features which that Strimzi provides.
+Everything is tested in a Kubernetes environment to fully represent a production or user environment.
+Again, to clarify I have an example:
+
+```java
+@Test
+void testReceiveSimpleMessage() {
+    KafkaTopicResource.topic(CLUSTER_NAME, TOPIC_NAME).done();
```

Comment:
I think you should add a comment to lines like this, because it's meaningless to someone who doesn't know our system tests.